File: src/components/JSONHandler/src/ALRPCObjectsImpl/V2/GPSData.cpp
#include "../include/JSONHandler/ALRPCObjects/V2/GPSData.h"
#include "GPSDataMarshaller.h"
#include "CompassDirectionMarshaller.h"
#include "DimensionMarshaller.h"
/*
  interface	Ford Sync RAPI
  version	2.0O
  date		2012-11-02
  generated at	Thu Jan 24 06:36:23 2013
  source stamp	Thu Jan 24 06:35:41 2013
  author	robok0der
*/



using namespace NsAppLinkRPCV2;

GPSData::GPSData(const GPSData& c)
{
  *this=c;
}


bool GPSData::checkIntegrity(void)
{
  return GPSDataMarshaller::checkIntegrity(*this);
}


GPSData::GPSData(void)
{
}



bool GPSData::set_actual(bool actual_)
{
  actual=actual_;
  return true;
}

bool GPSData::set_altitude(int altitude_)
{
  if(altitude_>10000)  return false;
  altitude=altitude_;
  return true;
}

bool GPSData::set_compassDirection(const CompassDirection& compassDirection_)
{
  if(!CompassDirectionMarshaller::checkIntegrityConst(compassDirection_))   return false;
  compassDirection=compassDirection_;
  return true;
}

bool GPSData::set_dimension(const Dimension& dimension_)
{
  if(!DimensionMarshaller::checkIntegrityConst(dimension_))   return false;
  dimension=dimension_;
  return true;
}

bool GPSData::set_hdop(unsigned int hdop_)
{
  if(hdop_>31)  return false;
  hdop=hdop_;
  return true;
}

bool GPSData::set_heading(unsigned int heading_)
{
  if(heading_>360)  return false;
  heading=heading_;
  return true;
}

bool GPSData::set_latitudeDegrees(int latitudeDegrees_)
{
  if(latitudeDegrees_>1000000000)  return false;
  latitudeDegrees=latitudeDegrees_;
  return true;
}

bool GPSData::set_longitudeDegrees(int longitudeDegrees_)
{
  if(longitudeDegrees_>1000000000)  return false;
  longitudeDegrees=longitudeDegrees_;
  return true;
}

bool GPSData::set_pdop(unsigned int pdop_)
{
  if(pdop_>31)  return false;
  pdop=pdop_;
  return true;
}

bool GPSData::set_satellites(unsigned int satellites_)
{
  if(satellites_>31)  return false;
  satellites=satellites_;
  return true;
}

bool GPSData::set_speed(unsigned int speed_)
{
  if(speed_>400)  return false;
  speed=speed_;
  return true;
}

bool GPSData::set_utcDay(unsigned int utcDay_)
{
  if(utcDay_>31)  return false;
  if(utcDay_<1)  return false;
  utcDay=utcDay_;
  return true;
}

bool GPSData::set_utcHours(unsigned int utcHours_)
{
  if(utcHours_>23)  return false;
  utcHours=utcHours_;
  return true;
}

bool GPSData::set_utcMinutes(unsigned int utcMinutes_)
{
  if(utcMinutes_>59)  return false;
  utcMinutes=utcMinutes_;
  return true;
}

bool GPSData::set_utcMonth(unsigned int utcMonth_)
{
  if(utcMonth_>12)  return false;
  if(utcMonth_<1)  return false;
  utcMonth=utcMonth_;
  return true;
}

bool GPSData::set_utcSeconds(unsigned int utcSeconds_)
{
  if(utcSeconds_>59)  return false;
  utcSeconds=utcSeconds_;
  return true;
}

bool GPSData::set_utcYear(unsigned int utcYear_)
{
  if(utcYear_>2100)  return false;
  if(utcYear_<2010)  return false;
  utcYear=utcYear_;
  return true;
}

bool GPSData::set_vdop(unsigned int vdop_)
{
  if(vdop_>31)  return false;
  vdop=vdop_;
  return true;
}




bool GPSData::get_actual(void) const
{
  return actual;
}


int GPSData::get_altitude(void) const
{
  return altitude;
}


const CompassDirection& GPSData::get_compassDirection(void) const 
{
  return compassDirection;
}


const Dimension& GPSData::get_dimension(void) const 
{
  return dimension;
}


unsigned int GPSData::get_hdop(void) const
{
  return hdop;
}


unsigned int GPSData::get_heading(void) const
{
  return heading;
}


int GPSData::get_latitudeDegrees(void) const
{
  return latitudeDegrees;
}


int GPSData::get_longitudeDegrees(void) const
{
  return longitudeDegrees;
}


unsigned int GPSData::get_pdop(void) const
{
  return pdop;
}


unsigned int GPSData::get_satellites(void) const
{
  return satellites;
}


unsigned int GPSData::get_speed(void) const
{
  return speed;
}


unsigned int GPSData::get_utcDay(void) const
{
  return utcDay;
}


unsigned int GPSData::get_utcHours(void) const
{
  return utcHours;
}


unsigned int GPSData::get_utcMinutes(void) const
{
  return utcMinutes;
}


unsigned int GPSData::get_utcMonth(void) const
{
  return utcMonth;
}


unsigned int GPSData::get_utcSeconds(void) const
{
  return utcSeconds;
}


unsigned int GPSData::get_utcYear(void) const
{
  return utcYear;
}


unsigned int GPSData::get_vdop(void) const
{
  return vdop;
}

File: mobile/android/SyncProxyAndroid/src/main/java/com/ford/syncV4/proxy/rpc/GPSData.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;

import com.ford.syncV4.proxy.RPCStruct;
import com.ford.syncV4.proxy.constants.Names;
import com.ford.syncV4.proxy.rpc.enums.CompassDirection;
import com.ford.syncV4.proxy.rpc.enums.Dimension;
import com.ford.syncV4.proxy.rpc.enums.DisplayType;
import com.ford.syncV4.util.DebugTool;

public class GPSData extends RPCStruct {

    public GPSData() { }
    public GPSData(Hashtable hash) {
        super(hash);
    }
    public void setLongitudeDegrees(Integer longitudeDegrees) {
    	if (longitudeDegrees != null) {
    		store.put(Names.longitudeDegrees, longitudeDegrees);
    	} else {
    		store.remove(Names.longitudeDegrees);
    	}
    }
    public Integer getLongitudeDegrees() {
    	return (Integer) store.get(Names.longitudeDegrees);
    }
    public void setLatitudeDegrees(Integer latitudeDegrees) {
    	if (latitudeDegrees != null) {
    		store.put(Names.latitudeDegrees, latitudeDegrees);
    	} else {
    		store.remove(Names.latitudeDegrees);
    	}
    }
    public Integer getLatitudeDegrees() {
    	return (Integer) store.get(Names.latitudeDegrees);
    }
    public void setUtcYear(Integer utcYear) {
    	if (utcYear != null) {
    		store.put(Names.utcYear, utcYear);
    	} else {
    		store.remove(Names.utcYear);
    	}
    }
    public Integer getUtcYear() {
    	return (Integer) store.get(Names.utcYear);
    }
    public void setUtcMonth(Integer utcMonth) {
    	if (utcMonth != null) {
    		store.put(Names.utcMonth, utcMonth);
    	} else {
    		store.remove(Names.utcMonth);
    	}
    }
    public Integer getUtcMonth() {
    	return (Integer) store.get(Names.utcMonth);
    }
    public void setUtcDay(Integer utcDay) {
    	if (utcDay != null) {
    		store.put(Names.utcDay, utcDay);
    	} else {
    		store.remove(Names.utcDay);
    	}
    }
    public Integer getUtcDay() {
    	return (Integer) store.get(Names.utcDay);
    }
    public void setUtcHours(Integer utcHours) {
    	if (utcHours != null) {
    		store.put(Names.utcHours, utcHours);
    	} else {
    		store.remove(Names.utcHours);
    	}
    }
    public Integer getUtcHours() {
    	return (Integer) store.get(Names.utcHours);
    }
    public void setUtcMinutes(Integer utcMinutes) {
    	if (utcMinutes != null) {
    		store.put(Names.utcMinutes, utcMinutes);
    	} else {
    		store.remove(Names.utcMinutes);
    	}
    }
    public Integer getUtcMinutes() {
    	return (Integer) store.get(Names.utcMinutes);
    }
    public void setUtcSeconds(Integer utcSeconds) {
    	if (utcSeconds != null) {
    		store.put(Names.utcSeconds, utcSeconds);
    	} else {
    		store.remove(Names.utcSeconds);
    	}
    }
    public Integer getUtcSeconds() {
    	return (Integer) store.get(Names.utcSeconds);
    }
    public void setCompassDirection(CompassDirection compassDirection) {
    	if (compassDirection != null) {
    		store.put(Names.compassDirection, compassDirection);
    	} else {
    		store.remove(Names.compassDirection);
    	}
    }
    public CompassDirection getCompassDirection() {
        Object obj = store.get(Names.compassDirection);
        if (obj instanceof CompassDirection) {
            return (CompassDirection) obj;
        } else if (obj instanceof String) {
        	CompassDirection theCode = null;
            try {
                theCode = CompassDirection.valueForString((String) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.compassDirection, e);
            }
            return theCode;
        }
        return null;
    }
    public void setPdop(Integer pdop) {
    	if (pdop != null) {
    		store.put(Names.pdop, pdop);
    	} else {
    		store.remove(Names.pdop);
    	}
    }
    public Integer getPdop() {
    	return (Integer) store.get(Names.pdop);
    }
    public void setHdop(Integer hdop) {
    	if (hdop != null) {
    		store.put(Names.hdop, hdop);
    	} else {
    		store.remove(Names.hdop);
    	}
    }
    public Integer getHdop() {
    	return (Integer) store.get(Names.hdop);
    }
    public void setVdop(Integer vdop) {
    	if (vdop != null) {
    		store.put(Names.vdop, vdop);
    	} else {
    		store.remove(Names.vdop);
    	}
    }
    public Integer getVdop() {
    	return (Integer) store.get(Names.vdop);
    }
    public void setActual(Boolean actual) {
    	if (actual != null) {
    		store.put(Names.actual, actual);
    	} else {
    		store.remove(Names.actual);
    	}
    }
    public Boolean getActual() {
    	return (Boolean) store.get(Names.actual);
    }
    public void setSatellites(Integer satellites) {
    	if (satellites != null) {
    		store.put(Names.satellites, satellites);
    	} else {
    		store.remove(Names.satellites);
    	}
    }
    public Integer getSatellites() {
    	return (Integer) store.get(Names.satellites);
    }
    public void setDimension(Dimension dimension) {
    	if (dimension != null) {
    		store.put(Names.dimension, dimension);
    	} else {
    		store.remove(Names.dimension);
    	}
    }
    public Dimension getDimension() {
        Object obj = store.get(Names.dimension);
        if (obj instanceof Dimension) {
            return (Dimension) obj;
        } else if (obj instanceof String) {
        	Dimension theCode = null;
            try {
                theCode = Dimension.valueForString((String) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.dimension, e);
            }
            return theCode;
        }
        return null;
    }
    public void setAltitude(Integer altitude) {
    	if (altitude != null) {
    		store.put(Names.altitude, altitude);
    	} else {
    		store.remove(Names.altitude);
    	}
    }
    public Integer getAltitude() {
    	return (Integer) store.get(Names.altitude);
    }
    public void setHeading(Integer heading) {
    	if (heading != null) {
    		store.put(Names.heading, heading);
    	} else {
    		store.remove(Names.heading);
    	}
    }
    public Integer getHeading() {
    	return (Integer) store.get(Names.heading);
    }
    public void setSpeed(Integer speed) {
    	if (speed != null) {
    		store.put(Names.speed, speed);
    	} else {
    		store.remove(Names.speed);
    	}
    }
    public Integer getSpeed() {
    	return (Integer) store.get(Names.speed);
    }
}
